### ISSUE TITLE
USWDS - Bug: Remove focus on text in validation component

### ISSUE THREAD

**Issue by amycole501** [uswds/uswds#5760]
### Describe the bug

When auditing the validation component for keyboard accessibility I noticed that the text inside the "legend" area above the input field was focusable. Each validation check received a visible focus and JAWS instructed me to "use the arrow keys" to toggle between the items in the list.

### Steps to reproduce the bug

1. Go to https://designsystem.digital.gov/components/validation/
2. Use a keyboard to tab into the component example
3. Note that each sentence/validation check receives focus
![image](https://github.com/uswds/uswds/assets/117778073/d1b779bf-71f3-427b-a51d-710521c51f61)
4. If you listen using a screen reader you may be prompted to toggle between the two using the arrow keys on your keyboard

### Expected Behavior

No visible focus should occur in the two checks/prompts. They look like text and it's an uncommon pattern to have focus appear around text without a clear indication of what the focus is supposed to do. Interacting with text seems like a bug.

### Related code

The checklist code seems to be causing this behavior
<li class="usa-checklist__item" data-validator="uppercase">
            Use at least one uppercase character
          </li>
          <li class="usa-checklist__item" data-validator="numerical">
            Use at least one number
          </li>

Recommend changing it from a list of two items to something that doesn't cause focus to be drawn to the text. 

### System setup

PC
Desktop/laptop
Chrome and Edge

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by mahoneycm** [comment]
So it looks like we intentionally added the ability to tab to these items in #4914 in an effort to make it easier for screen reader users to target the checklist items. 

> Added tabindex="0" to make the items in the validation checklist items easier to navigate via tab [29c2b61](https://github.com/uswds/uswds/pull/4914/commits/29c2b618e75eb6e8ec4727323835e93e1f9077dd)

@amycole501 Is there a concern with screen readers missing these items when tabbing to the input? Do you see any benefit in having them targeted by the tab?  Or is it too distracting and breaking expected convention?

**Comment by amycole501** [comment]
Yes because text typically isn't focusable in the DOM unless it's interactive. It could prove confusing. Will follow up with some suggestions including some from Jacline's user research with visually impaired testers.

**Comment by amycole501** [comment]
Following up with some further context: If text is focusable usually it needs to be interactive and perform a function. 

Instead of the instructions text becoming focusable, I recommend that when a person tab into the text field, the helper text should act as form field instructions within the label. 
https://www.w3.org/WAI/tutorials/forms/instructions/#providing-instructions-within-labels

For example, if the data entry field receives focus, the screen reader would say "Code requirements: Use at least one uppercase character. Use at least one number." Then once the person has typed in an uppercase character or a number, they hear (as it works now) "complete".
